app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,81 @@
+class InvitationsController < ApplicationController
+  before_action :restrict_access_on_create, only: [ :new, :create ]
+
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params)
+    invitation_token = SecureRandom.urlsafe_base64
+    @invitation.digest = digest invitation_token
+
+    if (@invitation.save)
+      # Send mail
+      UserMailer.invitation(invitation_token, @invitation.email).deliver
+      redirect_to action: :new
```

Comment:
Well, we should redirect user somewhere after invitation was sent. Thought of it as of a best option.